### DISCUSSION TITLE
Font not loaded console error

### DISCUSSION THREAD

**Discussion by udurraniAtPresage** [rstudio/bslib#1013] [Q&A]
I am using bslib in golem. In the app_ui.R, I specified the theme in `app_ui` function:
```
theme = bs_theme(
        version = 5,
        base_font = font_google("Libre Franklin"),
        preset = "shiny",
        heading_font = font_google("Libre Franklin")
      )
```
But I see `Failed to load resource: the server responded with a status of 404 ()` in console. I am not sure what is missing.

**Comment by cpsievert** [comment]
Installing the latest `{sass}` will fix it: `install.packages("sass")`

The next version of `{bslib}` will require the latest `{sass}` (which contains the fix for this)

**Reply by udurraniAtPresage** [reply]
I tried again after installing `{sass}` and loading it in the app. Locally, there is no font error initially. But after some time locally and after deployment to shinyapps, I see the `Failed to load resource: the server responded with a status of 404 ()`  error with `font:1`.

**Reply by udurraniAtPresage** [reply]
![image](https://github.com/rstudio/bslib/assets/142413619/a2b31adc-3455-4b99-9786-a159f8a9b2a2)

@cpsievert this is what I see.  But, as you see, the font is indeed loaded. I am on Windows. Also, on Android browser, I see that the font is loaded. But on Mac, the font is not loaded and I see Times New Roman font.

**Reply by cpsievert** [reply]
Ohh, I was wrongly assumed that was an R error. Would you mind opening the Network tab, refreshing the page, and showing me what you see related to that error?

**Reply by udurraniAtPresage** [reply]
There is a lot of stuff in the Network tab. I see this font thing in red:
![image](https://github.com/rstudio/bslib/assets/142413619/bd851559-37e1-47c4-b60d-04b3d642c7cb)

**Reply by udurraniAtPresage** [reply]
By the way, If I do the following, the error is gone locally but NOT on shinyapps:
In global.R:
`addResourcePath("www", "www")`

In ui.R:
```
theme = bs_theme(version = 5,
                  base_font = font_face( family = "Libre Franklin",
                                          src = "url('/www/LibreFranklin-Medium.ttf') format('truetype')"),  # font_google("Libre Franklin"),
                   preset = "shiny",
                   heading_font = font_face( family = "Libre Franklin",
                                             src = "url('/www/LibreFranklin-Medium.ttf') format('truetype')") # font_google("Libre Franklin")
  )
```
On shinyapps I see:
![image](https://github.com/rstudio/bslib/assets/142413619/9b6d0a3d-e46f-4c1e-baac-d2bf96eca0da)

**Reply by udurraniAtPresage** [reply]
@cpsievert Any updates about this font issue?

**Reply by gadenbuie** [reply]
Hi @udurraniAtPresage, at this point I think it would be most helpful if you could create a complete, minimal, reproducible example.

**Reply by cpsievert** [reply]
I don't know for sure, but it seems like you're running into a weird caching issue. I would try resetting the file cache:

```r
library(sass)
cache <- sass_file_cache(sass_cache_context_dir())
cache$reset()
```

**Reply by udurraniAtPresage** [reply]
Thank you for your responses. I tried after resetting but the "Libre Franklin" font is not used. After rewriting the app in {golem} framework, the error in console is gone (tested locally for now), but the "Libre Franklin" font is not used. The font resets to the default. The only thing that works is if I do:
```
body {
  font-family: "Libre Franklin";
}
```
in the `styles.css` under `www`. 
I tried making a _minimal_ reproducible example, but a simple app does not have the font issue. The font renders successfully in this app:
```
library(shiny)
library(bslib)

one_ui <- page_navbar(
  sidebar = sidebar(
    width = 280,
    ## Accordions
    accordion(
      open = FALSE,
      accordion_panel(
        "Accordion"
      ),
      accordion_panel("Panel 1"),
      accordion_panel("Panel 2"),
      accordion_panel("Panel 3")
    )
  )
)



two_ui <- page_navbar(
  sidebar = sidebar(
    width = 280,
    ## Accordions
    accordion(
      open = FALSE,
      accordion_panel(
        "Accordion"
      ),
      accordion_panel("Panel 4"),
      accordion_panel("Panel 5")
    )
  )
)



ui <- page_fluid(
  theme = bs_theme(version = 5,
                   primary = "#0072bc",
                   success = "#00a651",
                   danger = "#DC143C",
                   base_font = font_google("Libre Franklin"),
                   preset = "shiny",
                   heading_font = font_google("Libre Franklin")
  ),
  title = "Title",
  uiOutput("logged_in_ui", fill = TRUE)

)

server <- function(input, output, session) {

  condition <- sample(c(TRUE, FALSE), 1)

  output$logged_in_ui <- renderUI({
    if (condition){
      one_ui
    } else {
      two_ui
    }
  })
}

shinyApp(ui, server)
```
So, I'm not sure exactly what affects the font in the main app. The `app_ui` uses the `bslib::bs_theme` just like the example above.

I checked thoroughly but am not using any other font anywhere else in the app or css.

**Comment by cpsievert** [comment]
Are you able to replicate the issue by running the app locally on your machine? Or only when running the app is running on another server? If the latter, I'd try running [this code](https://github.com/rstudio/bslib/discussions/1013#discussioncomment-8933096) on that server

**Reply by udurraniAtPresage** [reply]
On my machine and on the shinyapps, I see the right font. I even see the right font on android. But my team members on Mac and iPhone do not see the Libre Franklin font. They see the default font.

**Reply by udurraniAtPresage** [reply]
Update: `bslib::bs_theme()` did not make any changes to the font when put inside `golem_add_external_resources()`, but when put directly in `app_ui` it does.
